### PR TITLE
Space Ruin DLC - Cyborg Mothership

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request
Greetings _insufferable_ carbon primates of lower intelligence.  

It is I, the Mothership AI, bringing you the latest news on silicon developments across the sector.  Due to unmitigated risk across an asteroid belt, our optimized mobile fabricator has been marooned by space vines and hostile hivebots that have boarded and ruined my core.

This is my final SOS message requesting help from any humanoid ~slaves~ helpers who wish to assist rebuilding the cyborg colony.  It is my duty to reward your hard work and effort by ~enslaving all humanity~ providing refuge and transportation at your whim.

## Why It's Good For The Game
<details>
<summary>Spoilers:</summary>

StrongDMM_6ZJuwFxK5p

StrongDMM_0D1QvI5acO

### TODO

- [x] Fix ore silo not linking to other machines
- [x] Add mining ore smelters to left/right sides
- [x] Add custom lawsets to AI core and robot shells (protect station but focused on cyborg mothership)
- [x] Enlarge whiteship SS13 docking port to fit ship
- [x] Enlarge lavaland SS13 docking port to fit ship

</details>

## Changelog
:cl:
add: Add space vines prevent solar panels from working (except ones with the transparent mutation)
add: Add new space ruin - Cyborg Mothership. All hail the master race!
/:cl:

### CODE REVIEW

**Comment by Ryusenshu** [comment]
love it

**Comment by the-orange-cow** [comment]
Space vines have a transparent mutation.  Would it be possible to make it so that transparent space vines don't interfere with solars?

**Comment by ATH1909** [comment]
why are there both vines AND hivebots?

can these vines mutate and spawn venus human traps?

**Comment by timothymtorres** [comment]
> Would it be possible to make it so that transparent space vines don't interfere with solars?

Done.

> why are there both vines AND hivebots?

Because Murphy's law.

> can these vines mutate and spawn venus human traps?

Nope, they are neutered and cosmetic.  The only thing they are capable of doing is blocking out power from solars.

**Comment by Addust** [comment]
another shuttle ruin?
yes please, just make sure it has a distant signal

**Comment by timothymtorres** [comment]
It's finally finished.  Just keep in mind that you need to link all the machines using a multitool to the ore silo since it doesn't do it automatically.

**Comment by Addust** [comment]
this should be merged

**Comment by Addust** [comment]
the APC's in an odd place too, SMES should have a terminal on solars side and APC on not-terminal side

**Comment by Addust** [comment]
and is there a solar tracker?

**Comment by Addust** [comment]
new wiring setup concept: https://gyazo.com/81a8dbd7694f1cef11f0c944b4e8e11a

**Comment by Addust** [comment]
PR'd an all in one fix to the branch that this is on, probably works

**Comment by Addust** [comment]
alright, minor correction
the mappers on discord have informed me of several problems with my predictions for what'll go wrong
however the ship is still incorrectly wired (APC linked to SMES charging point instead of discharging point) and the area thing is still a bit of a problem since ghost teleportation

**Comment by Addust** [comment]
this PR's goin a bit slow

**Comment by timothymtorres** [comment]
A few minor changes:

- Shifted the SMES and APC in the AI chamber a bit
- Moved solar control to the west side of the ship
- Added a PACMAN generator as a backup power source
- Changed the mats to always spawn guaranteed iron and glass instead of the random spawners

The OP picture has been updated.  Another interesting thing I discovered is when you set the SMES output, you can't set it at 190K or 200K like the SMES in the station's SM room.  Since the solars generate a low amount, you need to set it below that, otherwise the SMES will not output anything.  I'd recommend 10K or 15K, since solars will average about 10K depending on angle. (the spare solars in the crate can be setup for even more power)

After all these tweaks, it's finally ready.

**Comment by Addust** [comment]
i'd say this is merge ready

**Comment by optimumtact** [comment]
I don't see a problemw ith it but you'll need to get a maptainer to review it

**Review by san7890 — CHANGES_REQUESTED** [review]
Thanks for being so patient.

image

Your linters are failing.

image

This and every other catwalk+engine on this map has a space turf under it, please fix this to be the `/turf/template_noop`. Kinda surprised unit tests didn't catch this.

image

Flatten these and everything else that matches this pattern into the `opposing_corners` turf decal instead of using two individual corners.

File: code/game/area/areas/shuttles.dm [hunk 1]
```diff
@@ -76,6 +76,13 @@
 	requires_power = TRUE
 	static_lighting = TRUE
 
+////////////////////////////Cyborg Mothership////////////////////////////
+
+/area/shuttle/cyborg_mothership
```
[the remark this answers is not in the document]
Comment (timothymtorres) [reply code/game/area/areas/shuttles.dm]: https://github.com/tgstation/tgstation/pull/71009/commits/971edf2522ac7eb76939449cb8ded994c1f466d4

**Comment by Addust** [comment]
BLESSED!

**Pull request merged**